### PR TITLE
Added Exception and logging

### PR DESCRIPTION
**Pull request by uckamdar59**
Do not merge right now

### CODE REVIEW

**Review by uckamdar59 — COMMENTED** [review]
Do not merge right now!

**Review by uckamdar59 — COMMENTED** [review]
check and merge

File: src/main/java/com/TruckApi/TruckApi/Service/TruckServiceImpl.java [hunk 1]
```diff
@@ -126,9 +114,10 @@ else if ("STANDARD_CONTAINER".equals(String.valueOf(truckRequest.getTruckType())
 		TruckTransporterData sData = new TruckTransporterData();
 		sData.setTransporterId(truckRequest.getTransporterId());
 		sData.setTruckId(truckId_temp);
-
+
```
Comment (badireddisatyasaivarun) [inline src/main/java/com/TruckApi/TruckApi/Service/TruckServiceImpl.java]: check this secondary truck_transporter table too, like adding @NotBlank in its entities

File: src/main/java/com/TruckApi/TruckApi/Service/TruckServiceImpl.java [hunk 2]
```diff
@@ -222,43 +224,55 @@ else if ("HIGH_CUBE_CONTAINER".equals(String.valueOf(truckUpdateRequest.getTruck
...
-			return truckDeleteResponse;
-		} else {
-			truckDao.delete(findTruckData);
-			sTruckDao.delete(findTruckTransporterData);
+		TruckData truckData = truckDao.findByTruckId(id);
+		TruckTransporterData findTruckTransporterData = sTruckDao.findByTruckId(id);
 
-			truckDeleteResponse.setStatus(truckConstants.DELETE_SUCCESS);
-			return truckDeleteResponse;
+		if (truckData == null || findTruckTransporterData == null) {
```
Comment (badireddisatyasaivarun) [inline src/main/java/com/TruckApi/TruckApi/Service/TruckServiceImpl.java:241]: but we don't know from which table, the error is coming from, so better write them in two individual " if " conditions

File: src/main/java/com/TruckApi/TruckApi/Service/TruckServiceImpl.java [hunk 3]
```diff
@@ -273,18 +287,22 @@ public TruckData getDataById(String Id) {
...
 				return truckDao.findByTruckApproved(truckApproved, currentPage);
 			}
 
 			else if (transporterId != null && truckApproved != null) {
+				log.info("Truck Data with params returned");
 				return truckDao.findByTransporterIdAndTruckApproved(transporterId, truckApproved, currentPage);
 			}
 
 		}
+		log.info("Truck Data with params returned");
```
Comment (badireddisatyasaivarun) [inline src/main/java/com/TruckApi/TruckApi/Service/TruckServiceImpl.java:304]: the statement is wrong, though the method is with params, we are not sending get all with params..

**Review by liveasy97 — APPROVED** [review]

**Pull request merged**